src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataReaderExtensions.cs

```diff
@@ -71,7 +60,26 @@ public static ODataItemBase ReadResourceOrResourceSet(this ODataReader reader)
                             "The resource which is ending should be on the top of the items stack.");
                         itemsStack.Pop();
                         break;
+                    case ODataReaderState.DeletedResourceStart:
+                        ODataDeletedResource deletedResource = (ODataDeletedResource)reader.Item;
+                        ODataResourceWrapper deletedResourceWrapper = null;
+                        Contract.Assert(deletedResource != null, "Deleted resource should not be null");
 
+                        deletedResourceWrapper = new ODataResourceWrapper(deletedResource);
```

Comment:
Move line 66 (Contract.Assert) to mmediately follow assignment of deletedResource in line 64 -- that's what it corresponds to.

combine lines 65 and 68 into:

                        ODataResourceWrapper deletedResourceWrapper =  new ODataResourceWrapper(deletedResource)